Update development instructions

tijmenb opened the pull request:
- Add setup / running the app instructions and comply with other readme guidelines from the styleguides. 
- Add a license.
- Run app on 0.0.0.0. This allows us to access the running application inside a VM
  (dev.gov.uk:5000).

rboulton commented on README.md:
````diff
@@ -7,3 +7,39 @@ it in action at https://govuk-content-explorer.herokuapp.com/
+
+```
+source ENV/bin/activate
+ENV=dev ./start.py
````
Perhaps suggest the "-d" flag here, since it's usually wanted when running in dev.

tijmenb committed:
Update development instructions

- Add setup / running the app instructions and comply with other readme
guidelines from the styleguides.
- Add a license.
- Run app on 0.0.0.0. This allows us to access the running application
inside a VM (dev.gov.uk:5000).

tijmenb commented:
Done - rebased & squashed.
